vault/resource_kv_secret_test.go

```diff
@@ -21,13 +23,25 @@ func TestAccKVSecret(t *testing.T) {
 		PreCheck:  func() { testutil.TestAccPreCheck(t) },
 		Steps: []resource.TestStep{
 			{
-				Config: testKVSecretConfig(mount, name),
+				Config: testKVSecretConfig_basic(mount, name),
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr(resourceName, consts.FieldPath, fmt.Sprintf("%s/%s", mount, name)),
 					resource.TestCheckResourceAttr(resourceName, "data.zip", "zap"),
 					resource.TestCheckResourceAttr(resourceName, "data.foo", "bar"),
 				),
 			},
+			{
+				Config: testKVSecretConfig_basic(mount, name),
+				Check:  testResourceKVSecret_apiAcessCheck,
+			},
+			{
+				Config: testKVSecretConfig_updated(mount, name),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, consts.FieldPath, fmt.Sprintf("%s/%s", mount, name)),
+					resource.TestCheckResourceAttr(resourceName, "data.zip", "zoop"),
+					resource.TestCheckResourceAttr(resourceName, "data.foo", "bar"),
```

Comment:
Add extra API check function.

Reply:
I hope, I understand this correctly: I also added an extra API check test to the updated test.

Reply:
yes, that is correct.

Reply:
done with [add extra api check to update test](https://github.com/hashicorp/terraform-provider-vault/pull/1570/commits/4347da9e0ac956801a41c0fc82a890042fe58041)